packages/eslint-plugin/tests/rules/require-types-exports.test.ts

```diff
@@ -0,0 +1,1985 @@
+      export const f = (a: A = 1): void => {};
+    `,
+
+    `
+      export type A = number;
+      export function f(): A {}
+    `,
+
+    `
+      export type A = number;
+      export const f = (): A => {};
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export function f(): A | B {}
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export const f = (): A | B => {};
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export function f(): A & B {}
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export const f = (): A & B => {};
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export function f(): [A, B] {}
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export const f = (): [A, B] => {};
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export function f(): { a: A; b: B } {}
+    `,
+
+    `
+      export type A = number;
+      export type B = string;
+      export const f = (): { a: A; b: B } => {};
+    `,
+
+    `
+      import { testFunction, type Arg } from './module';
+
+      export function f(a: Arg): void {}
+    `,
+
+    `
+      import { Arg } from './types';
+
+      export function f(a: Arg): void {}
+    `,
+
+    `
+      import type { Arg } from './types';
+
+      export function f(a: Arg): void {}
+    `,
+
+    `
+      import type { ImportedArg as Arg } from './types';
+
+      export function f(a: Arg): void {}
+    `,
+
+    `
+      import type { Arg } from './types';
+
+      export function f<T extends Arg>(a: T): void {}
+    `,
+
+    `
+      export type R = number;
+
+      export function f() {
+        const value: { num: R } = {
+          num: 1,
+        };
+
+        return value;
+      }
+    `,
+
+    `
+      import type { A } from './types';
+
+      export type T1 = number;
+
+      export interface T2 {
+        key: number;
+      }
+
+      export const value: { a: { b: { c: T1 } } } | [string, T2 | A] = {
+        a: {
+          b: {
+            c: 1,
+          },
+        },
+      };
+    `,
+
+    `
+      import type { A } from './types';
+
+      export type T1 = number;
+
+      export interface T2 {
+        key: number;
+      }
+
+      const value: { a: { b: { c: T1 } } } | [string, T2 | A] = {
+        a: {
+          b: {
+            c: 1,
+          },
+        },
+      };
+
+      export default value;
+    `,
+
+    `
+      export enum Fruit {
+        Apple,
+        Banana,
+        Cherry,
+      }
+
+      export function f(a: Fruit): void {}
+    `,
+
+    `
+      export function f(arg: Record<PropertyKey, Promise<string>>) {
+        return arg;
+      }
+    `,
+
+    `
+      export function f<T extends Record<PropertyKey, Promise<string>>>(arg: T) {
+        return arg;
+      }
+    `,
```

Comment:
Those 2 tests are failing for some reason. TBH, it was an accidental discovery.
Seems like `isSymbolFromDefaultLibrary(PropertyKey)` returns false for some reason. Am I doing something wrong?